.circleci/merge_base_branch.sh

```diff
@@ -0,0 +1,22 @@
+echo 'export CIRCLE_PR_NUMBER="${CIRCLE_PR_NUMBER:-${CIRCLE_PULL_REQUEST##*/}}"' >> $BASH_ENV
+source $BASH_ENV
+
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    # Update PR refs for testing.
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/head:pr/${CIRCLE_PR_NUMBER}/head"
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/merge:pr/${CIRCLE_PR_NUMBER}/merge"
+
+    # Retrieve the refs
+    git fetch -u origin ${FETCH_REFS}
+
+    # Checkout PR merge ref.
+    git checkout -qf "pr/${CIRCLE_PR_NUMBER}/merge"
```

Comment:
I *think* that conflicting PRs don't get a `/merge` ref, so this will fail, and the next command is a bit strange. what is your thinking here?

Reply:
I have no idea -- to be honest, this is just cargo culting. I haven't had any luck finding ANY documentation on the /merge ref, or the `pr/` refset at all on GitHub -- my google-fu has failed me unfortunately. This is just the method that was recommended on the CircleCI forums (because Circle doesn't have a "merge before run" functionality like Travis).